File: bounce-game.py
from random import randint
pos = randint(1, 10)
backwards = False

for i in range(1, 100):
    field = '!          !'
    field = field[:pos] + 'O' + field[pos+1:]
    if field[len(field)-2] == 'O':
        backwards = True
    elif field[1] == 'O':
        backwards = False
    print(field)

    if backwards:
        pos -= 1
    else: 
        pos += 1

File: bounce-game.swift
var pos = Int.random(in: 1...10)
var backwards = false

for _ in 1...100 {
  var field = "!          !"
  
  field = field.prefix(pos) + "O" + field.dropFirst(pos + 1)
  
  if(pos == field.count - 2){
    backwards = true
  } else if(pos == 1){
    backwards = false
  }
  print(field)
  
  if(backwards){
    pos -= 1;
  } else {
    pos += 1;
  }
}